File: src/java/cl/unab/cliquecover/Grafo.java
package cl.unab.cliquecover;

public class Grafo {
    private final int v; 
    private int e;       
    private final boolean[][] matrizAdyacencia;

    public Grafo(int v) {
        this.v = v;
        this.e = 0;
        this.matrizAdyacencia = new boolean[v][v];
    }

    public void agregarArista(int u, int w) {
        if (!matrizAdyacencia[u][w]) {
            matrizAdyacencia[u][w] = true;
            matrizAdyacencia[w][u] = true;
            this.e++;
        }
    }

    public boolean sonAdyacentes(int u, int w) {
        if (u == w) return true; 
        return matrizAdyacencia[u][w];
    }

    public int getV() {
        return v;
    }

    public int getE() {
        return e;
    }
}

File: src/python/grafo.py
class Grafo:
    def __init__(self, v):
        self.v = v
        self.e = 0
        self.matriz_adyacencia = [[False] * v for _ in range(v)]

    def agregar_arista(self, u, w):
        if not self.matriz_adyacencia[u][w]:
            self.matriz_adyacencia[u][w] = True
            self.matriz_adyacencia[w][u] = True
            self.e += 1

    def son_adyacentes(self, u, w):
        if u == w:
            return True
        return self.matriz_adyacencia[u][w]